c7n/resources/account.py

```diff
@@ -1839,3 +1839,37 @@ def process(self, resources, event=None):
         if state == bool(sechub):
             return resources
         return []
+
+
+@filters.register('lakeformation-cross-account')
+class LakeformationFilter(Filter):
+    """To check if S3 bucket i.e. registered for Lakeformation belongs to the same account or not.
+
+    :example:
+
+    .. code-block:: yaml
+
+       policies:
+         - name: lakeformation-cross-account-bucket
+           resource: aws.account
+           filters:
+            - type: lakeformation-cross-account
+
+    """
+
+    def process(self, resources, event=None):
+        Buckets = self.manager.get_resource_manager('s3').resources()
```

Comment:
Lakeformation can have multiple source data store resource types. This is only looking at S3 buckets. A name change on the filter to clarify that would make more sense imo or we could look at other resource types too?

Reply:
changed it !

Reply:
We are only looking for only s3 resources.